xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -126,6 +150,272 @@ define(["jquery"], function ($) {
 
 
 
+
+  /**
+   * Return the property descriptor corresponding to a property id
+   * @param {String} propertyId
+   * @returns {Object}
+   */
+  Logic.prototype.getPropertyDescriptor = function (propertyId) {
+    return this.data.meta.propertyDescriptors.find(function (descriptor) {
+      return descriptor.id === propertyId;
+    });
+  };
+
+
+
+  /**
+   * Return a new displayer based on the specified property and row data
+   * @param {String} propertyId The id of the property of the entry
+   * @param {Object} entry The entry data object
+   * @param {String} displayerId The
+   */
+  Logic.prototype.createDisplayer = function (propertyId, entry, displayerId) {
+    var self = this;
+    var defer = $.Deferred();
+
+    // default displayerId
+    if (displayerId === undefined) {
+      displayerId = ((this.getPropertyDescriptor(propertyId) || {}).displayer || {}).id || "default";
+    }
+
+    // load displayer based on it's id
+    require([BASE_PATH + "displayers/" + displayerId + "Displayer.js"],
+      // load success
+      function (Displayer) {
+        var displayer = new Displayer(propertyId, entry, self);
```

Comment:
Depends on what you want to resolve: the displayer creation, or the displayer view element creation.
For now it resolves to the displayer creation, because the displayer view creation belongs to another asynchronous operation.

Reply:
Yes, but since the other asynchronous operation is triggered inside the constructor I don't have access to its promise.